### PR TITLE
Make official v2 release + gh action for docker image

### PR DESCRIPTION
**Pull request by korikuzma**
@acoffman and @susannasiebert agreed that we should just make the official v2 release and release molecular profiles separately

Note: This assumes the GH release tag does **not** include the `v` prefix. For instance, use `1.0.0` rather than `v1.0.1`

### CODE REVIEW

**Comment by korikuzma** [comment]
Added work for creating docker images. I didn't pin a version in the Dockerfile. Wasn't sure the best way to handle this. I can revert back to manually changing or think of an alternate way without having to change the version in 2 places. 

I think we'll also want to set the secrets in the repo to use the griffithlab user or a different user?

**Comment by susannasiebert** [comment]
> Added work for creating docker images. I didn't pin a version in the Dockerfile. Wasn't sure the best way to handle this. I can revert back to manually changing or think of an alternate way without having to change the version in 2 places.

I think this is fine although we should monitor this in practice. Sometimes pypi's registry gets backed up and even though a new release is live, pip isn't aware of the new version so an old version would get installed/released in the Docker image.

> I think we'll also want to set the dockerhub secrets in the repo to use the griffithlab user or a different user?

That is a good question. I haven't set up a github action for this before so I'm not entirely sure. Let me ask some of my colleagues who might have more info.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM python:3.7
+
+RUN --mount=type=secret,id=civicpy_version \
+  civicpy_version="$(cat /run/secrets/civicpy_version)" \
+  && pip install civicpy==${civicpy_version}
```
Comment (korikuzma) [inline Dockerfile:5]: @susannasiebert Added these lines to install specific version based on the GH release tag. Exposing GH actions env var to Dockerfile was tricky. [This](https://github.com/docker/build-push-action/blob/master/docs/advanced/secrets.md) was one solution that worked. If you know of an alternate way, we can do that.

**Comment by korikuzma** [comment]
> I think this is fine although we should monitor this in practice. Sometimes pypi's registry gets backed up and even though a new release is live, pip isn't aware of the new version so an old version would get installed/released in the Docker image.

@susannasiebert I can see if there's a way to check that the version exists before starting docker build action.

**Review by susannasiebert — APPROVED** [review]
+1

**Pull request merged**